### PR TITLE
[IOS-579, IOS-574]: New chat controller and pushed search header

### PR DESCRIPTION
**Pull request by yuliaveres**
screenshot at apr 09 14-58-33

We do not have search related functionality yet.

### CODE REVIEW

File: Toshi/Controllers/Recent/NewChatViewController.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    private lazy var tableView: UITableView = {
+        let view = UITableView(frame: CGRect.zero, style: .plain)
+        view.translatesAutoresizingMaskIntoConstraints = false
+
+        view.backgroundColor = Theme.viewBackgroundColor
+        BasicTableViewCell.register(in: view)
+        view.delegate = self
+        view.dataSource = self
+        view.sectionFooterHeight = 0.0
```
Comment (designatednerd) [inline Toshi/Controllers/Recent/NewChatViewController.swift]: you can probably just use `.zero` for this

File: Toshi/Controllers/Recent/NewChatViewController.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    func searchHeaderWillBeginEditing(_ headerView: PushedSearchHeaderView) {
+        dataSource.isSearching = true
+    }
+
+    func searchHeaderWillEndEditing(_ headerView: PushedSearchHeaderView) {
+        dataSource.isSearching = false
+    }
+
+    func searchHeaderDidReceiveCancelEvent(_ headerView: PushedSearchHeaderView) {
+
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Recent/NewChatViewController.swift]: i will see if it is not needed, for now this implementation is without search results view being shown, so maybe something will be needed to do here with the table view..now this is just main tableview displaying 2 cells without search results tableview on top. search will be implemented in separate pr

**Comment by marijnschilling** [comment]
screenshot 2018-04-10 12 56 42

ooh in the sketch file i see the little 🔍-icon, do we need that in this PR?

File: Toshi/Controllers/Recent/NewChatViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (designatednerd) [reply Toshi/Controllers/Recent/NewChatViewController.swift]: Then at least throw in a //TODO or other comment indicating this is coming in a later PR plz

**Comment by marijnschilling** [comment]
In the transition from main chat to new chat the status bar acts a little weird
statusbar

**Comment by yuliaveres** [comment]
@marijnschilling  yes! could you branch-out and add icon ? maybe improve transition ? 🙏  thx

File: Toshi/Controllers/Recent/NewChatViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Recent/NewChatViewController.swift]: sure

**Comment by marijnschilling** [comment]
Yes I will add the changes of the icon! And look for the transition thing ✌️

File: Toshi/Controllers/Profiles/ProfilesViewController.swift [hunk 3]
```diff
@@ -107,7 +107,7 @@ final class ProfilesViewController: UIViewController {
 
         switch type {
         case .favorites:
-            controller.searchBar.placeholder = Localized.profiles_search_favorites_placeholder
+            break
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Profiles/ProfilesViewController.swift]: there is no more `.favorites` type for Profiles. will remove all related

[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Profiles/ProfilesViewController.swift]: or just merge 👍

File: Toshi/Controllers/Recent/UsersBotsGroupsDatasource.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    var limit: Int
+    var offset: Int
+    var type: String?
+
+    init(searchText: String = "", limit: Int = 100, offset: Int = 0, type: String? = nil, isSearching: Bool = false) {
+        self.searchText = searchText
+        self.limit = limit
+        self.offset = offset
+        self.type = type
+        self.isSearching = isSearching
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Recent/UsersBotsGroupsDatasource.swift]: it is not needed now.

File: Toshi/Controllers/Recent/NewChatViewController.swift [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard indexPath.row < newChatItems.count else { return UITableViewCell() }
+
+        let item = newChatItems[indexPath.row]
+        let cellData = TableCellData(title: item.title, leftImage: item.icon)
+
+        let configurator = CellConfigurator()
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: configurator.cellIdentifier(for: cellData.components), for: indexPath) as? BasicTableViewCell else { return UITableViewCell() }
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Recent/NewChatViewController.swift]: the `reuseIdentifier` is taken from `cellData`. this guard can't be *higher* ..............

File: Toshi/Controllers/Recent/NewChatViewController.swift (hunk 1 above)
Comment (yuliaveres) [reply Toshi/Controllers/Recent/NewChatViewController.swift]: type CGFloat has no member `zero`

File: Toshi/Generated/Code/LocalizedStrings.swift [hunk 6]
```diff
@@ -248,6 +248,10 @@ enum LocalizedKey: String, StringCaseListable {
     tab_bar_title_chats,
     tab_bar_title_dapps,
     tab_bar_title_favorites,
+<<<<<<< HEAD
```
Comment (marijnschilling) [inline Toshi/Generated/Code/LocalizedStrings.swift]: oei, some conflict marker slipped in here

**Pull request merged**